### PR TITLE
fix: `Message.getInfo`

### LINKED ISSUE #2567: Method "Message.getInfo()" return error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'sendQueryMsgInfo')
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When I send a message with "client.sendMessage" and when I return, I call the "getInfo" method, it always returns this error message: "Evaluation failed: TypeError: Cannot read properties of undefined (reading 'sendQueryMsgInfo')"

### Expected behavior

The expectation was that it would return the following object:
"{
         delivery: Array<{id: ContactId, t: number}>,
         deliveryRemaining: number,
         played: Array<{id: ContactId, t: number}>,
         playedRemaining: number,
         read: Array<{id: ContactId, t: number}>,
         readRemaining: number
     }"

### Steps to Reproduce the Bug or Issue

1. Get a session
2. Send a message, with the "client.sendMessage" method
3. upon return, call the "getInfo()" method

### Relevant Code

if (messageType == 'message')
      client.sendMessage(number, mensagem).then(async response => {
        const info = await response.getInfo()
        if(haveDeliveryRemaining(info))
          sendMessageWithOut9(client,number,mensagem)
      }).catch(err => {
        console.log("Erro! ao enviar mensagem, descrição: ", err);
      });

 function haveDeliveryRemaining (info) {
    return info.deliveryRemaining >= 1
  }
  function sendMessageWithOut9(client, number, mensagem){
    client.sendMessage(number.replace('9',''), mensagem).then(async response => {}).catch(err => {
      console.log("Erro! ao enviar mensagem, descrição: ", err);
    });
  }

### Browser Type

Google Chrome

### WhatsApp Account Type

WhatsApp Business

### Does your WhatsApp account have multidevice enabled?

No, I am not using Multi Device

### Environment

- OS: Windows 11

- Phone OS: Android 12

- whatsapp-web.js version: https://github.com/Julzk/whatsapp-web.js/tarball/jkr_hotfix_8

- Node js version: 18.17.1

### Additional context

_No response_

### LINKED ISSUE #2575: Error: Evaluation failed: TypeError: r.findImpl is not a function
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Here in Brazil +55, depending on each region of the country, some WhatsApp numbers have the prefix 9 and others do not, when I send the message I validate using "MessageInfo" if it was delivered, if it was not delivered I remove the prefix "9" and I resend the message, upon resending, the following error is returning to me:

```
Error: Evaluation failed: TypeError: r.findImpl is not a function
    at https://web.whatsapp.com/app.1b6002989bf96f16e110.js:55:173854
    at Generator.next (<anonymous>)
    at t (https://web.whatsapp.com/vendor1~app.6470964955d882fdfd6f.js:2:66483)
    at s (https://web.whatsapp.com/vendor1~app.6470964955d882fdfd6f.js:2:66694)
    at https://web.whatsapp.com/vendor1~app.6470964955d882fdfd6f.js:2:66753
    at new Promise (<anonymous>)
    at https://web.whatsapp.com/vendor1~app.6470964955d882fdfd6f.js:2:66634
    at M._serverQuery (https://web.whatsapp.com/app.1b6002989bf96f16e110.js:55:174182)
    at M._query (https://web.whatsapp.com/app.1b6002989bf96f16e110.js:55:173465)
    at M.find (https://web.whatsapp.com/app.1b6002989bf96f16e110.js:55:171949)
```

### Expected behavior

I expected the message to be sent

### Steps to Reproduce the Bug or Issue

1 - Get a session
2- Send a message, with the "client.sendMessage" method, `with number: 55<dd><8 digit number>`

Example with success
`55<dd>9<8 digit number>`

Example with error
`55<dd><8 digit number>`

### Relevant Code

 ```
client.sendMessage('5574*********@c.us', mensagem).then(async response => {
        const info = await response.getInfo()
        if(haveDeliveryRemaining(info))
          sendMessageWithOut9(sender,number,mensagem)
      }).catch(err => {
        console.log("Erro! ao enviar mensagem, descrição: ", err);
      });
  function haveDeliveryRemaining (info) {
    return info.deliveryRemaining >= 1
  }
  function sendMessageWithOut9(client, number, mensagem){
    client.sendMessage(number.replace('9',''), mensagem).then(async response => {}).catch(err => {
      console.log("Erro! ao enviar mensagem, descrição: ", err);
    });
  }
```

### Browser Type

Google Chrome

### WhatsApp Account Type

WhatsApp Business

### Does your WhatsApp account have multidevice enabled?

No, I am not using Multi Device

### Environment

OS: Windows 11

Phone OS: Android 12

whatsapp-web.js version: github:alechkos/whatsapp-web.js#fix-getInfo

Node js version: 18.17.1

### Additional context

Arose from the resolution of this problem:
https://github.com/pedroslopez/whatsapp-web.js/issues/2567

### PR DESCRIPTION
**Pull request by alechkos**
# Table of Contents

### - [How Has the PR Been Tested (latest test on 29.10.2023)](https://github.com/pedroslopez/whatsapp-web.js/pull/2574#tests)
- [**The environment the PR was tested on**](https://github.com/pedroslopez/whatsapp-web.js/pull/2574#env)

### - [Types of Changes](https://github.com/pedroslopez/whatsapp-web.js/pull/2574#types-of-changes)

---

## Description <a href="https://github.com/pedroslopez/whatsapp-web.js/pull/2574#description" id="description"/>
In latest versions wa refactored `sendQueryMsgInfo` method which caused an issue, the PR fixes that issue.

---

## Related Issues <a href="https://github.com/pedroslopez/whatsapp-web.js/pull/2574#related-issues" id="related-issues"/>
PR fixes #2567 fixes #2575

---

## To test this PR by yourself you can run one of the following commands: <a href="https://github.com/pedroslopez/whatsapp-web.js/pull/2574#pr-testing" id="pr-testing"/>

- **NPM**
```powershell
npm install github:alechkos/whatsapp-web.js#fix-getInfo
```
- **YARN**
```powershell
yarn add github:alechkos/whatsapp-web.js#fix-getInfo
```

---

## If you encounter any errors while testing this PR, please provide in a comment: <a href="https://github.com/pedroslopez/whatsapp-web.js/pull/2574#pr-testing-issue" id="pr-testing-issue"/>

1. The code you've used without any sensitive information (use [syntax highlighting](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting) for more readability)
2. The error you got
3. The library version
4. The WWeb version: `console.log(await client.getWWebVersion());`
5. The browser (Chrome/Chromium)

> [!IMPORTANT]
> **You have to [reapply](https://github.com/pedroslopez/whatsapp-web.js/pull/2574#pr-testing) the PR each time it is changed (new commits were pushed since your last application)**

---

## How Has The PR Been Tested (latest test on 29.10.2023) <a href="https://github.com/pedroslopez/whatsapp-web.js/pull/2574#tests" id="tests"/>
Was tested with a code below:
```js
client.on('message_create', async (message) => {
    console.log(await (await client.sendMessage(message.to, 'TEST')).getInfo());
});
```

The response after the message was read by one participant:
```js
{
  delivery: [],
  deliveryRemaining: 1,
  played: [],
  playedRemaining: 2,
  read: [
    {
      id: [
        {
          server: 'c.us',
          user: 'XXXXXXXXXX',
          _serialized: 'XXXXXXXXXX@c.us'
        }
      ],
      t: 169XXXXXXX
    }
  ],
  readRemaining: 1
}
```

#### Types of accounts:
- Personal
- Buisness

#### Environment:
- Android 10
- Windows 10:
    - WWebJS **v1.23.1-alpha.0**
    - WWeb **v2.2346.52**, **v2.2347.56**
    - Puppeteer **v18.2.1**
    - Node **v18.17.1**
    - Chrome **v117.0.5938.150**

---

## Types of Changes <a href="https://github.com/pedroslopez/whatsapp-web.js/pull/2574#types-of-changes" id="types-of-changes"/>
- [ ] Dependency change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix/feature that would cause existing functionality to change)

## Checklist
- [X] My code follows the code style of this project
- [ ] I have updated the usage example accordingly (example.js)
- [ ] I have updated the documentation accordingly (index.d.ts)

### CODE REVIEW

**Review by CSFelix — APPROVED** [review]
Tested it locally and worked!!

**Comment by N0N4M3BNS** [comment]
If using web version 2.2344.53 not work. Can check it?

**Review by N0N4M3BNS — APPROVED** [review]

**Review by N0N4M3BNS — APPROVED** [review]
Ok working now.

File: src/structures/Message.js [hunk 1]
```diff
@@ -534,15 +534,18 @@ class Message extends Base {
...
-     * Get information about message delivery status. May return null if the message does not exist or is not sent by you.
+     * Get information about message delivery status.
+     * May return null if the message does not exist or is not sent by you.
      * @returns {Promise<?MessageInfo>}
      */
     async getInfo() {
         const info = await this.client.pupPage.evaluate(async (msgId) => {
             const msg = window.Store.Msg.get(msgId);
-            if (!msg) return null;
+            if (!msg?.fromMe) return null;
```
Comment (tuyuribr) [inline src/structures/Message.js:544]: This will break node 13

**Review by tuyuribr — CHANGES_REQUESTED** [review]

**Review by tuyuribr — APPROVED** [review]
Good work

**Comment by wilsinho8** [comment]
Im testing and i got this error: 

image

**Comment by alechkos** [comment]
@wilsinho8 
> Im testing and i got this error

It's very easy to READ the PR description
Written specially for you lol:
https://github.com/pedroslopez/whatsapp-web.js/pull/2574#pr-testing-issue

**Review by shirser121 — APPROVED** [review]
Thanks!

**Pull request merged**